### PR TITLE
Move originpolymorphihelpers, describer and imageutils from pkg/oc/lib to staging

### PR DESCRIPTION
**Pull request by soltysh**
This builds on top of https://github.com/openshift/origin/pull/23020 and https://github.com/openshift/origin/pull/23093, so only 4 last commits matter.

### CODE REVIEW

File: pkg/oc/lib/describe/describer.go [hunk 1]
```diff
@@ -685,14 +687,21 @@ func DescribeImage(image *imagev1.Image, imageName string) (string, error) {
 			formatAnnotations(out, image.ObjectMeta, "")
 		}
 
+		if err := ocimageutil.ImageWithMetadata(image); err != nil {
+			return err
+		}
+		dockerImage, ok := image.DockerImageMetadata.Object.(*dockerv10.DockerImage)
+		if !ok {
+			return fmt.Errorf("unable to read image metadata")
```
Comment (mfojtik) [inline pkg/oc/lib/describe/describer.go]: nit: misleading... we were able to read image metadata, but we were not able to convert them to appropriate object? probably showing what is the source will help debug this :)

File: staging/src/github.com/openshift/oc/pkg/helpers/graph/genericgraph/test/new-project-deployed-app.yaml [hunk 2]
```diff
@@ -543,7 +543,7 @@ items:
       name: frontend
     tls:
       insecureEdgeTerminationPolicy: Redirect
-      type: Edge
+      termination: edge
```
Comment (mfojtik) [inline staging/src/github.com/openshift/oc/pkg/helpers/graph/genericgraph/test/new-project-deployed-app.yaml]: are you sure about lowercase?

File: pkg/oc/lib/describe/describer_test.go [hunk 3]
```diff
@@ -87,10 +82,13 @@ var MissingDescriberCoverageExceptions = []reflect.Type{
 }
 
 func TestDescriberCoverage(t *testing.T) {
+	scheme := runtime.NewScheme()
+	kubernetesscheme.AddToScheme(scheme)
+	api.Install(scheme)
```
Comment (mfojtik) [inline pkg/oc/lib/describe/describer_test.go]: nit: util.Must()

**Comment by mfojtik** [comment]
/lgtm
/hold 

(hold for real bump)

File: pkg/oc/lib/describe/describer_test.go (hunk 3 above)
Comment (soltysh) [reply pkg/oc/lib/describe/describer_test.go]: These are tests so it'll fail one way or the other, I don't care that much.

File: staging/src/github.com/openshift/oc/pkg/helpers/graph/genericgraph/test/new-project-deployed-app.yaml (hunk 2 above)
Comment (soltysh) [reply staging/src/github.com/openshift/oc/pkg/helpers/graph/genericgraph/test/new-project-deployed-app.yaml]: Yeah, double checked with API, it's not the type I care here but the termination. Previously we relied on defaulting to kick in, now these tests need to get object as if it came from the server.

File: pkg/oc/lib/describe/describer.go (hunk 1 above)
Comment (soltysh) [reply pkg/oc/lib/describe/describer.go]: Will fix.

**Pull request merged**